src/analysis/modals/AnalysisModal.ts

```diff
@@ -261,15 +261,17 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
       GALAXY: currentApp(appToolLabels.GALAXY),
       CROMWELL: currentApp(appToolLabels.CROMWELL),
       HAIL_BATCH: currentApp(appToolLabels.HAIL_BATCH),
+      WDS: currentApp(appToolLabels.WDS),
     };
 
     const appDisabledMessages: Record<AppToolLabel, string> = {
       GALAXY: 'You already have a Galaxy environment',
       CROMWELL: 'You already have a Cromwell instance',
       HAIL_BATCH: 'You already have a Hail Batch instance',
+      WDS: 'You already have a Workspace Data Service instance',
```

Comment:
where is this actually used? We do not currently enforce anything about multiple WDSes in your workspace, so this message is misleading.

( we don't expose the ability to create multiple WDSes, but users could do it by making direct API calls to Leo )

Reply:
It looks like the analysis modal doesn't support launching more than one instance of any app. But WDS won't show up as an option in the analysis modal because it's not present in the `cloudAppTools` constants in tool-utils.
https://github.com/DataBiosphere/terra-ui/blob/16f1083f6407a1979b38a016ecd9a128cc7a368a/src/analysis/utils/tool-utils.ts#L140-L143

Ideally, the key type for this Record would not be `AppToolLabel`, but rather a union of the tools in `cloudAppTools`.

Reply:
I think we can improve these types after upgrading TypeScript (#4066) and getting access to the `satisfies` operator.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html

Reply:
Resolved in https://github.com/DataBiosphere/terra-ui/pull/4060/commits/5a4c7631eac14e72b5836819c318b168d1bf5d3c.